cfme/configure/settings.py

```diff
@@ -184,273 +172,372 @@ def delete(self, cancel=False, *time_objs):
-        long_text = BootstrapSelect("quad_truncate")
-
-    @View.nested
-    class visualdisplay(View):   # noqa
-        chart_theme = BootstrapSelect("display_reporttheme")
-        time_zone = BootstrapSelect("display_timezone")
-
-    save = Button("Save")
-    reset = Button("Reset")
-
+class DefaultViews(Updateable, NavigatableMixin):
+    # Basic class for navigation to default views screen
+    look_up = {
+        'Flavors': ['clouds', 'flavors'],
+        'Instances': ['clouds', 'instances'],
+        'Availability Zones': ['clouds', 'availability_zones'],
+        'Images': ['clouds', 'images'],
+        'Cloud Providers': ['clouds', 'providers'],
+        'Stacks': ['clouds', 'stacks'],
+        'Compare': ['general', 'compare'],
+        'Compare Mode': ['general', 'compare_mode'],
+        'Infrastructure Providers': ['infrastructure', 'infrastructure_providers'],
+        'Configuration Management Providers': ['infrastructure',
+                                               'configuration_management_providers'],
+        'VMs': ['infrastructure', 'vms'],
+        'My Services': ['services', 'my_services'],
+        'Catalog Items': ['services', 'catalog_items'],
+        'Templates & Images': ['services', 'templates'],
+        'VMs & Instances': ['services', 'vms_instances'],
+        'Containers Providers': ['containers', 'providers'],
+        'Nodes': ['containers', 'nodes'],
+        'Pods': ['containers', 'pods'],
+        'Services': ['containers', 'services'],
+        'Routes': ['containers', 'routes'],
+        'Containers': ['containers', 'containers'],
+        'Projects': ['containers', 'projects'],
+        'Replicators': ['containers', 'replicators'],
+        'Container Images': ['containers', 'images'],
+        'Image Registries': ['containers', 'image_registries'],
+        'Builds': ['containers', 'builds'],
+        'Volumes': ['containers', 'volumes'],
+        'Templates': ['containers', 'templates']
+    }
+
+    def __init__(self, appliance, my_settings):
+        self.appliance = appliance
+        self.my_settings = my_settings
+
+    def set_default_view(self, button_group_names, defaults, fieldset=None):
+        """This function sets default views for the objects.
 
-@navigator.register(Visual, 'All')
-class VisualAll(CFMENavigateStep):
-    prerequisite = NavigateToAttribute('appliance.server', 'MySettings')
-    VIEW = VisualTabForm
+        Args:
+            button_group_names: either the name of the button_group_name
+                                or list of the button groups to set the
+                                default view for.
+            defaults: the default view to set. in case that button_group_names
+                     is a list, you can either set 1 view and it'll be set
+                     for all the button_group_names or you can use a list
+                     (default view per button_group_name).
+        """
```

Comment:
Add 'Raises' to docblock for the AssertionError that happens when incorrect length lists are passed.